db/model/roomType.js

```diff
@@ -123,34 +123,41 @@ module.exports = function roomType(seq, dataTypes) {
             resolve(inst);
           }
 
-          inst._previousDataValues.bots.forEach((botName) => {
-            seq.models.Bot.findOne({
-              where: {
-                name: {
-                  $iLike: botName,
-                },
-              },
-            })
-            .then((o) => {
-              inst.removeBots(o)
-              .catch(reject);
-            });
-          });
-
-          inst.dataValues.bots.forEach((botName) => {
-            seq.models.Bot.findOne({
-              where: {
-                name: {
-                  $iLike: botName,
+          new Promise((resolve, reject) => {
+            if (inst._previousDataValues.bots) {
+              inst._previousDataValues.bots.forEach((botName) => {
+                seq.models.Bot.findOne({
+                  where: {
+                    name: {
+                      $iLike: botName,
+                    },
+                  },
+                })
+                .then((o) => {
+                  inst.removeBots(o)
+                  .catch(reject);
+                });
+              });
+            }
+
+            resolve();
```

Comment:
@iamigo that "resolve()" is for a promise inside the promise that is being returned. I've tested it and I've gotten past the "then" on 145.